### PR TITLE
Replace Cassandra visibility TTL with explicit DELETE

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Replace Cassandra visibility TTL with explicit `DELETE`.

**Why?**
Previously visibility records were deleted only when retention kicked in. For new `DeleteWorklfowExecution` API it is required to be able to delete visibility record in any given moment of time.

**How did you test it?**
Modify existing unit test. Will add integration tests for `DeleteWorkflowExecution` API in corresponding PR.

**Potential risks**
~~If Cassandra based visibility is used and workflow has many executions, new implementation is slower. No production environments should use Cassandra based visibility though.~~
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/persistence/visibility/store/standard/cassandra/visibility_store.go [hunk 1]
```diff
@@ -460,8 +437,47 @@ func (v *visibilityStore) ListClosedWorkflowExecutionsByStatus(
 	return response, nil
 }
 
-// DeleteWorkflowExecution is a no-op since deletes are auto-handled by cassandra TTLs
-func (v *visibilityStore) DeleteWorkflowExecution(_ *manager.VisibilityDeleteWorkflowExecutionRequest) error {
+func (v *visibilityStore) DeleteWorkflowExecution(request *manager.VisibilityDeleteWorkflowExecutionRequest) error {
+	// Primary key in closed_execution table has close_time which is not in the request.
+	// Query closed_execution table first (using secondary index by workflow_id) to get close_time.
+	// This is not very efficient if workflow has multiple executions.
```
Comment (yiminc) [inline common/persistence/visibility/store/standard/cassandra/visibility_store.go:443]: I don't like this part, this is one extra read for every visibility delete. Did you try to put close_time into task?

File: common/persistence/visibility/store/standard/cassandra/visibility_store.go [hunk 2]
```diff
@@ -460,8 +437,47 @@ func (v *visibilityStore) ListClosedWorkflowExecutionsByStatus(
...
+		Query(templateGetClosedWorkflowExecutionsByID,
+			request.NamespaceID.String(),
+			namespacePartition,
+			persistence.UnixMilliseconds(minTime),
+			persistence.UnixMilliseconds(maxTime),
+			request.WorkflowID).
+		Consistency(v.lowConslevel)
+	iter := query.Iter()
+
+	wfExecution, has := readClosedWorkflowExecutionRecord(iter)
```
Comment (yiminc) [inline common/persistence/visibility/store/standard/cassandra/visibility_store.go:454]: Does this query handles pagination as well? If workflowID is a cron, you could have many RunIDs.

File: common/persistence/visibility/store/standard/cassandra/visibility_store.go (hunk 1 above)
Comment (alexshtin) [reply common/persistence/visibility/store/standard/cassandra/visibility_store.go]: Changed.

File: common/persistence/visibility/store/standard/cassandra/visibility_store.go (hunk 2 above)
Comment (alexshtin) [reply common/persistence/visibility/store/standard/cassandra/visibility_store.go]: Not relevant anymore, but it turned out that default page size is 5000 (not unlimited, as I would expect it to be if not specified).

**Review by yiminc — APPROVED** [review]

**Pull request merged**